cli_tools/gce_windows_upgrade/upgrader/workflows.go

```diff
@@ -505,6 +506,12 @@ func (u *upgrader) runWorkflowWithSteps(workflowName string, timeout string, pop
 	}
 
 	setWorkflowAttributes(w, u)
+	defer func() {
+		// reuse the same logger across all workflows
+		if u.Logger == nil {
+			u.Logger = w.Logger
+		}
```

Comment:
Why is this done both here and in generateWorkflowWithSteps()?

Reply:
one is u.logger=w.Logger, another is w.Logger=u.logger.
I removed if-condition in generateWorkflowWithSteps in the new commit to simplify the logic, and improved comment to avoid confusion.